src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React,
+{
+  useState,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { getTodos, getVisibleTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+
+  const getTodosFromApi = async () => {
+    try {
+      const todosFromApi = await getTodos();
+
+      setTodos(todosFromApi);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromApi();
+  }, []);
+
+  const visibleTodos = getVisibleTodos(todos, query, filterBy);
+
+  const changeSelectedTodo = useCallback((id: number) => {
+    setSelectedTodoId(id);
+  }, []);
+
+  const activeTodo = useMemo(() => (
+    todos.find(({ id }) => id === selectedTodoId)
+  ), [selectedTodoId]);
```

Comment:
You can save the selected todo instead of saving id to avoid finding this todo every time